File: Irrgarten_Ruby/textUI.rb

require 'io/console'
require_relative 'Irrgarten'

module UI

  class TextUI

    #https://gist.github.com/acook/4190379
    def read_char
      STDIN.echo = false
      STDIN.raw!
    
      input = STDIN.getc.chr
      if input == "\e" 
        input << STDIN.read_nonblock(3) rescue nil
        input << STDIN.read_nonblock(2) rescue nil
      end
    
    
      return input
    end

    def next_move
      print "Where? "
      got_input = false
      while (!got_input)
        c = read_char
        case c
          when "\e[A"
            puts "UP ARROW"
            output = Directions::UP
            got_input = true
          when "\e[B"
            puts "DOWN ARROW"
            output = Directions::DOWN
            got_input = true
          when "\e[C"
            puts "RIGHT ARROW"
            output = Directions::RIGHT
            got_input = true
          when "\e[D"
            puts "LEFT ARROW"
            output = Directions::LEFT
            got_input = true
          when "\u0003"
            puts "CONTROL-C"
            got_input = true
            exit(1)
          else
            #Error 
        end
      end
      output
    end

    def show_game(game_state)
	puts game_state.players
      	puts "\n"
     	puts game_state.monsters
      	puts "\n"
      	puts game_state.labyrinth
      	puts "\n"

      	puts "Log:\n" + game_state.log

      	if (game_state.winner)
        	puts "Player " + game_state.current_player.to_s + "\n" + "ha ganado el juego"
      	else
        	puts "Current player: " + game_state.current_player.to_s
      	end 
    end

  end # class   

end # module   

File: Irrgarten_Java/Irrgarten_P3/src/irrgarten/UI/TextUI.java

package irrgarten.UI;

import irrgarten.Directions;
import irrgarten.GameState;
import java.util.Scanner;


public class TextUI {
    
    private static Scanner in = new Scanner(System.in);
    
    private char readChar() {
        String s = in.nextLine();     
        return s.charAt(0);
    }
    

    public Directions nextMove() {
        System.out.print("Where? ");
        
        Directions direction = Directions.DOWN;
        boolean gotInput = false;
        
        while (!gotInput) {
            char c = readChar();
            switch(c) {
                case 'w':
                    System.out.print(" UP\n");
                    direction = Directions.UP;
                    gotInput = true;
                    break;
                case 's':
                    System.out.print(" DOWN\n");
                    direction = Directions.DOWN;
                    gotInput = true;
                    break;
                case 'd':
                    System.out.print("RIGHT\n");
                    direction = Directions.RIGHT;
                    gotInput = true;
                    break;
                case 'a':
                    System.out.print(" LEFT\n");
                    direction = Directions.LEFT;
                    gotInput = true;    
                    break;
                
                default:
                    System.out.print("Please choose a valid character: w,a,s,d \n");
            }
        }    
        return direction;
    }
    
    public void showGame(GameState gameState) { 
        System.out.print(gameState.getPlayers() + "\n");
        System.out.print(gameState.getMonsters() + "\n");
        System.out.print("Log:\n" + gameState.getLog() + "\n");
        System.out.print(gameState.getLabyrinth() + "\n");
        if (gameState.getWinner()) {
            System.out.print("Player " + gameState.getCurrentPlayer() + " ha ganado el juego, ¡Felicidades! \n");
        }else{
            System.out.print("Current player: " + gameState.getCurrentPlayer() + "\n");
        }
    }
    
}
